src/state.rs

```diff
@@ -459,6 +476,36 @@ impl Common {
         let time = self.clock.now();
         let throttle = Some(Duration::from_secs(1));
 
+        if let Some(session_lock) = self.session_lock.as_ref() {
+            if let Some(lock_surface) = session_lock.surfaces.get(output) {
+                send_frames_surface_tree(
+                    lock_surface.wl_surface(),
+                    output,
+                    time,
+                    Some(Duration::ZERO),
+                    surface_primary_scanout_output,
+                );
+                if let Some(feedback) =
+                    source_node_for_surface(lock_surface.wl_surface(), &self.display_handle)
+                        .and_then(|source| dmabuf_feedback(source))
+                {
+                    send_dmabuf_feedback_surface_tree(
+                        &lock_surface.wl_surface(),
+                        output,
+                        surface_primary_scanout_output,
+                        |surface, _| {
+                            select_dmabuf_feedback(
+                                surface,
+                                render_element_states,
+                                &feedback.render_feedback,
+                                &feedback.scanout_feedback,
+                            )
+                        },
+                    )
+                }
+            }
```

Comment:
I guess given the fixed output we can also easily send the fractional scale here. (See the window code a few lines below.)